### PR TITLE
[TECHNICAL] Upgrade min SDK to Android 7.0 (v24)

### PR DESCRIPTION
**Pull request by Aitorbp**
## Related Issues
App:https://github.com/owncloud/android/issues/4230

- [x] Added changelog files for the fixed issues in folder changelog/unreleased. More info [here](https://github.com/owncloud/android/tree/master/changelog#create-changelog-items)
_____

### CODE REVIEW

File: owncloudApp/src/main/java/com/owncloud/android/presentation/documentsprovider/cursors/FileCursor.kt [hunk 1]
```diff
@@ -55,10 +54,8 @@ class FileCursor(projection: Array<String>?) : MatrixCursor(projection ?: DEFAUL
             flags = flags or Document.FLAG_DIR_SUPPORTS_CREATE
         }
 
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
-            flags = flags or Document.FLAG_SUPPORTS_COPY
-            flags = flags or Document.FLAG_SUPPORTS_MOVE
-        }
+        flags = flags or Document.FLAG_SUPPORTS_COPY
+        flags = flags or Document.FLAG_SUPPORTS_MOVE
```
Comment (JuancaG05) [inline owncloudApp/src/main/java/com/owncloud/android/presentation/documentsprovider/cursors/FileCursor.kt:58]: You can put this all together with the flags that don't require condition checks, the ones for delete and rename

**Review by JuancaG05 — CHANGES_REQUESTED** [review]
One tiny change here and ready to go! @Aitorbp

**Review by JuancaG05 — CHANGES_REQUESTED** [review]
We can safely remove the `ShareLinkToDialog.java` class as well, since it is not used anymore

**Review by JuancaG05 — APPROVED** [review]
LGTM

**Comment by jesmrec** [comment]
Some basic checks performed: installation, not installation in Android 6 and, video preview and share sheet.

Everything OK from my side to move forward.

**Pull request merged**